src/nft/hooks/usePriceImpact.ts

```diff
@@ -0,0 +1,46 @@
+import { Currency, Percent, TradeType } from '@uniswap/sdk-core'
+import { useMemo } from 'react'
+import { InterfaceTrade } from 'state/routing/types'
+import { useTheme } from 'styled-components/macro'
+import { computeRealizedPriceImpact, getPriceImpactWarning } from 'utils/prices'
+
+export interface PriceImpact {
+  percent: Percent
+  warning?: 'warning' | 'error'
+  toString(): string
+  warningColor?: string
+}
+
+export function usePriceImpact(trade?: InterfaceTrade<Currency, Currency, TradeType>): PriceImpact | undefined {
+  const theme = useTheme()
+
+  return useMemo(() => {
+    const marketPriceImpact = trade ? computeRealizedPriceImpact(trade) : undefined
+    const priceImpactWarning = marketPriceImpact ? getPriceImpactWarning(marketPriceImpact) : undefined
+    const warningColor =
+      priceImpactWarning === 'error'
+        ? theme.accentCritical
+        : priceImpactWarning === 'warning'
+        ? theme.accentWarning
+        : undefined
```

Comment:
nit: the warning and warning color are inherently tied, you could define a PriceImpactWarning interface with fields: `type` and `color` which is returned from your `getPriceImpactWarning` helper function.